### ISSUE TITLE
New DataTable clickable rows

### ISSUE THREAD

**Issue by joshxg** [openmsupply/mobile#1145]
## Is your feature request related to a problem? Please describe.

Rows need to be clickable in order to navigate

## Describe the solution you'd like

`Row` component should have an `onPress` prop. 

I think from a UX perspective it is potentially a source of confusing when rows look more or less exactly the same, but some are clickable and some aren't (current implementation). Some pages now use a column with an icon for clicking (stock take batches).

I think it's best to be consistent with this and add an icon to every row that is clickable. Potentially have the whole row be clickable, but also have an icon to indicate that it is clickable.

## Describe alternatives you've considered

Just use the current implementation

## Additional context

N/A

**Comment by joshxg** [comment]
Just had a look at this and it's a little more involved than first thought when you consider the `EditableCell` component. Having an `EditableCell` will trigger both the `Row.onPress` and the `EditableCell.onPress` which converts it into a `TextInput`. 

Looking into trying to capture the gesture event in the child and cancel it bubbling to the parent, at the moment

**Comment by joshxg** [comment]
From what I can find, preventing non-synthetic events from bubbling isn't possible in react-native, so there would have to be another solution involving wrapping individual cells in a `Touchable`, rather than the `Row`. I think it's going a bit far so will just wrap `Row` with a touchable and as an `onPress` prop. Would be nice to have an added icon if an `onPress` was supplied, but since the `Row` only is passed the `renderCell` method, it has to be done there.

**Comment by Chris-Petty** [comment]
It'd be great to actually survey some users if this would be better for them. But I think your rational is probably solid. My biggest concern is that on certain pages, column realestate is already scarce. 

I'd expect if you went the icon approach you'd need to do it in `renderCells`. Unless you did something like:
```javascript
const Row = React.memo(({ rowData, rowState, rowKey, renderCells, style }) => {
  const cells = useMemo(() => renderCells(rowData, rowState, rowKey), [rowData, rowState, rowKey]);
  if (!!onPress) cells.push(<Icon andWhatever />); // Lol mutation. You could avoid but this is just an example
  return <View style={style}>{cells}</View>;
});
```
All that said, I'm not fan yet. Doesn't the old implementation use a wrapping TouchableComponent?

**Comment by joshxg** [comment]
It basically does the same as this which isn't a problem as I don't believe there are any editable, but clickable rows. Possibly for expansions, but they only had editable-quantity-like cells, using a `TextInput` which is different than using a `Touchable` in a cell. Since we aren't using expansions, I don't think this will be a problem

**Comment by Chris-Petty** [comment]
Fixed in #1146
